src/test-apps/happy/tests/standalone/wdmNext/weave_wdm_next_test_base.py

```diff
@@ -135,106 +135,68 @@ def weave_wdm_next_test_base(self, wdm_next_args):
             self.__process_result("node01", "node02", value, data)
             self.result_data = data
 
+    def __validate_log_line(self, data, node_type, wdm_stress_check, stress_failures):
+        """
+        check expected log lines showed up in node logs with expected count
+        """
+        if node_type is "client":
+            node_log = data["client_output"]
+            log_check_list = self.client_log_check
+        else:
+            node_log = data["server_output"]
+            log_check_list = self.server_log_check
+        for expect_line_count in log_check_list:
+            item_output_cnt = len(re.findall(expect_line_count[0], node_log))
+            if item_output_cnt != expect_line_count[1]:
+                wdm_stress_check = False
+                error = "missing '%s' in %s_output, need %d, actual %d" % (expect_line_count[0], node_type, expect_line_count[1], item_output_cnt)
+                stress_failures.append(error)
+        return (wdm_stress_check, stress_failures)
+
+    def __validate_node_events(self, wdm_stress_check, stress_failures, data, client_gen=False, server_gen=False):
```

Comment:
This seems like a failure-prone API: exactly one of `client_gen` and `server_gen` MUST be true or the method will fail.  It would be better if the API took a single parameter, e.g. `client_gen`, and lines 162..173 were a `if...else` statement.  The calls to the __validate_node_events below will also need adjusting.

Reply:
def __validate_node_events(self, wdm_stress_check, stress_failures, data, use_client_gen):
    ...
    if use_client_gen:
        client_output = data["client_output"]
        ....
    else 
        client_out = data["server_output"]

Reply:
Thanks RS for the feedback!
It is much clear after the discussion, I will make changes now.

Reply:
done